storage-proofs/src/drgraph.rs

```diff
@@ -114,22 +117,28 @@ impl<H: Hasher> ParameterSetIdentifier for BucketGraph<H> {
 
 impl<H: Hasher> Graph<H> for BucketGraph<H> {
     #[inline]
-    fn parents(&self, node: usize) -> Vec<usize> {
+    fn parents(&self, node: usize, parents: &mut [usize]) {
         let m = self.base_degree;
 
         match node {
             // Special case for the first node, it self references.
-            0 => vec![0; m as usize],
             // Special case for the second node, it references only the first one.
-            1 => vec![0; m as usize],
+            0 | 1 => {
+                // Use the degree of the curren graph (`m`), as parents.len() might be bigger
+                // than that (that's the case for ZigZag Graph)
+                #[allow(clippy::needless_range_loop)]
+                for k in 0..m {
+                    parents[k] = 0;
+                }
```

Comment:
You should be able to do `for k in parents.iter_mut().take(m)` that way clippy is happy